internal/kubernetes/operator/resources/normalizer.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+// NormalizeAtlasResourceName normalizes the name to be used as a resource name in Kubernetes
+// Run fuzzing test if you want to change this function!
+func NormalizeAtlasResourceName(name string) string {
+	dictionary := map[string]string{
+		" ": "-",
+		".": "dot",
+		"@": "at",
+		"(": "left-parenthesis",
+		")": "right-parenthesis",
+		"&": "and",
+		"+": "plus",
+		":": "colon",
+		",": "comma",
+		"'": "single-quote",
+	}
```

Comment:
let's define this outside the function so it doesn't need to be recreated every-time the function is invoked

Reply:
Done